### PR TITLE
Skip link, Announce context consumer stories  added

### PR DESCRIPTION
**Pull request by mrebb**
Grommet stories added to storybook for SkipLinks and AnnounceContext.Consumer components

#### What does this PR do?
Added stories to Skip links and Announce Context.Consumer components on storybook

#### Where should the reviewer start?
SkipLinks.stories.js, announce-context.stories.js

#### What testing has been done on this PR?
storybook

#### How should this be manually tested?
storybook

#### What are the relevant issues?
#2647 

#### Do the grommet docs need to be updated?
yes

#### Should this PR be mentioned in the release notes?
yes

#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by mrebb** [comment]
@ShimiSun I implemented `SkipLink` example but placed the story under `SkipLinks`. There are two different components `SkipLink` and `SkipLinks`. I will move this example to SkipLink.

File: src/js/contexts/AnnounceContext/announce-context.stories.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+const AnnounceContextComponent = props => (
+  <Grommet theme={grommet} full>
+    <Box
+      justify="center"
+      align="center"
+      pad="xlarge"
+      background="linear-gradient(102.77deg, #865ED6 -9.18%, #18BAB9 209.09%)"
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/announce-context.stories.js]: I think this level of extra detail can be a little distracting, as the reader is drawn to it and away from the primary focus of the AnnounceContext. Is it needed?

File: src/js/contexts/AnnounceContext/announce-context.stories.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { storiesOf } from '@storybook/react';
+import { grommet } from 'grommet/themes';
+import { Box, Grommet, Text, AnnounceContext } from 'grommet';
+
+class Announcer extends Component {
+  static propTypes = {
+    announce: PropTypes.func.isRequired,
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/announce-context.stories.js]: Any reason not to include message, role, and politenessSetting?

File: src/js/contexts/AnnounceContext/announce-context.stories.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  constructor(props) {
+    super(props);
+    const { message, role, politenessSetting } = this.props;
+    this.state = {
+      message:
+        message || 'Here is a simple announcement. This will soon disappear',
+      politenessSetting: politenessSetting || 'polite',
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/announce-context.stories.js]: Can we align with the function parameter and call this `mode`?

Just as an aside, all props are "settings" of some sort, so the suffix "Setting" seems extraneous.

File: src/js/contexts/AnnounceContext/announce-context.stories.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      role: role || 'log',
+    };
+  }
+
+  componentDidMount() {
+    const { announce } = this.props;
+    const { message, politenessSetting } = this.state;
+    const timeout = 3000;
+    announce(message, politenessSetting, timeout);
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/announce-context.stories.js]: Can we update the AnnounceContext/doc.js file to mention the arguments to the `announce()` function?

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 3 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Sure, `mode` is perfect. Thanks!!

File: src/js/contexts/AnnounceContext/announce-context.stories.js [hunk 5]
```diff
@@ -2,35 +2,32 @@ import React, { Component } from 'react';
...
-    this.state = {
-      message:
-        message || 'Here is a simple announcement. This will soon disappear',
-      politenessSetting: politenessSetting || 'polite',
-      role: role || 'log',
-    };
+  
+  static defaultProps = {
+    message: 'Here is a simple announcement. This will soon disappear',
+      mode: 'polite',
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/announce-context.stories.js]: indentation is off, how did this make it through `prettier`?

File: src/js/contexts/AnnounceContext/doc.js [hunk 6]
```diff
@@ -13,7 +13,11 @@ export const doc = AnnounceContext => {
   DocumentedAnnounceContext.propTypes = {
     children: PropTypes.func.description(
       `Render function that will be called with an 'announce' function that
-      can be called when something should be announced.`,
+      can be called when something should be announced. 
+      'announce' function accepts 'message', 'mode' and 'timeout' as arguments
+      and these arguments can be passed as 'props' to the return component.
+      Example:  {announce => <Announcer announce={announce} {...props} />}
```
Comment (ericsoderberghp) [inline src/js/contexts/AnnounceContext/doc.js]: I think having `Announcer` here makes it look like a grommet component. How about something like:
```
{announce => <Button onClick={() => announce("Button was clicked", "polite", 1000)}
```
I think we should also mention possible values for `mode` and the units for `timeout`.

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 1 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Addressed in recent commit. Thanks !!

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 2 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Addressed in recent commit. Thanks !!

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 4 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Addressed in recent commit. Thanks !!

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 3 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Addressed in recent commit. Thanks !!

File: src/js/contexts/AnnounceContext/doc.js (hunk 6 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/doc.js]: Updated docs.

File: src/js/contexts/AnnounceContext/announce-context.stories.js (hunk 5 above)
Comment (mrebb) [reply src/js/contexts/AnnounceContext/announce-context.stories.js]: Not sure. Fixed it.

**Comment by ericsoderberghp** [comment]
I think the AnnounceContext stuff is good to go.

Regarding SkipLinks, I agree with @ShimiSun that we shouldn't have an example of SkipLink outside of the context of SkipLinks, as they are designed to go together.

Perhaps we can split this PR into two?

**Comment by mrebb** [comment]
Moved SkipLinks work to different branch and will make a different PR for the same.

**Pull request merged**